unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs

```diff
@@ -93,11 +121,37 @@ internal void Update(float deltaTime)
          * instead we should have a provider so all the subsystems can use it
          */
         float distanceToCamera = Vector3.Distance(cameraPosition, gameObject.transform.position);
-        float resolvedAlpha = forceShow ? TARGET_ALPHA_SHOW : ResolveAlphaByDistance(alpha, distanceToCamera, forceShow);
-        UpdateVisuals(resolvedAlpha);
         ScalePivotByDistance(distanceToCamera);
         LookAtCamera(cameraRight, cameraRotation.eulerAngles);
         pivot.transform.localPosition = Vector3.up * GetPivotYOffsetByDistance(distanceToCamera);
+
+        float resolvedAlpha = forceShow ? TARGET_ALPHA_SHOW : ResolveAlphaByDistance(alpha, distanceToCamera, forceShow);
+        if (ChechHide(resolvedAlpha))
+            return;
+
+        SetRenderersVisible(true);
+        float resolvedAlphaStep = GetNearestAlphaStep(resolvedAlpha);
+        float canvasAlphaStep = GetNearestAlphaStep(canvasGroup.alpha);
+
+        if (resolvedAlphaStep != canvasAlphaStep)
+            UpdateVisuals(resolvedAlpha);
+
+
+        // Local Methods
+        bool ChechHide(float checkAlpha)
```

Comment:
Not only this has a typo but the name also tells me that you are checking if something is hidden if alpha is < MINIMUM_ALPHA_TO_SHOW. This method also has the side effect of setting the renderer off if the condition is met.
We can give it a more proper name like `GetAndUpdateVisibility`

Reply:
Applying your commenet and merging after tests ^^